samples/textcompletion/csharp-ooproc/TextCompletions.cs

```diff
@@ -20,11 +21,13 @@ public static class TextCompletions
     /// and embeds it into a text prompt, which is then sent to the OpenAI completions API.
     /// </summary>
     [Function(nameof(WhoIs))]
-    public static string WhoIs(
+    public static HttpResponseData WhoIs(
         [HttpTrigger(AuthorizationLevel.Function, Route = "whois/{name}")] HttpRequestData req,
         [TextCompletionInput("Who is {name}?")] TextCompletionResponse response)
     {
-        return response.Content;
+        HttpResponseData responseData = req.CreateResponse(HttpStatusCode.OK);
+        responseData.WriteString(response.Content);
+        return responseData;
     }
 
     /// <summary>
```

Comment:
@cgillum , I had confusion with Generic completion, how can I create HttpResponseData object below since we do not pass the HttpRequestData object.

Reply:
There are two options we can consider:
1. Change the function signature to take an `HttpRequestData` instead of a `PromptPayload` or
2. Add a `FunctionsContext` parameter to the function and then use it to manually create an `HttpResponseData` object via its constructor - something like `new HttpResponseData(context);`

I'm not sure which option is the best but it may depend on the behavior of the binding expression rules. The second option might be the safest since it doesn't mess with the input binding behavior at all.

Reply:
used option 1